parse_nucleotide_fasta.c

```diff
@@ -0,0 +1,263 @@
+                argument_legality = 0;
+                flags[2].flag_raised = 1;
+                strcpy(flags[2].flag_value, optarg);
+                printf("Determining all k-mers of size %s.\n", optarg);
+                break;
+            case 'm':
+                argument_legality = 0;
+                flags[3].flag_raised = 1;
+                strcpy(flags[3].flag_value, optarg);
+                printf("Searching for all instances of %s in file.\n", optarg);
+                break;
+            case 'h':
+                printf("Parse_nuc_fasta Options: \n");
+                printf("\t-help : displays this message.\n");
+                printf("\t-file <file> : Directs the program to the fasta file.\n");
+                printf("\t-gc : global gc count.\n");
+                printf("\t-gc=<x> : gc count per x bases.\n");
+                printf("\t-k_mers <k> : lists all k-mers of size k.\n");
+                printf("\t-match <string> : searches for all instances of a string. (only words for same case searches)\n");
+                argument_legality = -1;
+                break;
+            case '?':
+                printf("Unknown or incorrect argument entered, exiting program. Please see maunal or enter ./parse_nuc_fasta -help\n");
+                argument_legality = -1;
+                break;
+            default:
+                printf("No legal options entered. Please see maunal or enter ./parse_nuc_fasta -help\n");
+                argument_legality = -1;
+                break;
+        }
+    }
+    if (flags[0].flag_raised == 0) {
+        printf("A fasta file must be passed in. Please see maunal or enter ./parse_nuc_fasta -help\n");
+        argument_legality = -1;
+    } else if (flags[1].flag_raised == 0 &&
+               flags[2].flag_raised == 0 &&
+               flags[3].flag_raised == 0) {
+        printf("No process arguments entered, select a process for the program carry out. Please see maunal or enter ./parse_nuc_fasta -help\n");
+        argument_legality = -1;
+    }
+    return argument_legality;
+}
+
+/******************************
+*   Counts the GC content in the read passed into the function. 
+*******************************/
+int GC_count(struct Flags* flags, char* read, int* site_values, int chunk_num, int chunk_val) {
+    // counts GC content per X bases
+    int site_index = 0;
+    while (read[site_index] != '\0') {
+        if(site_values[4] == chunk_val && chunk_val != 0) {
+            printf("Chunk number: %d A: %d T: %d C: %d G:%d\n", chunk_num, site_values[0], site_values[1], site_values[2], site_values[3]);
+            printf("Chunk: %d GC content: %f\n", chunk_num, (((float)site_values[2] + (float)site_values[3]) / (float)site_values[4]) * 100);
+            int i;
+            for (i = 0; i < 5; i++) {
+                site_values[i] = 0;
+            }
+            chunk_num++;
+        }
+        char site = read[site_index];
+        if (site == 'A' || site == 'a') {
```

Comment:
Avoid doing these double comparisons if ya can. Just convert to uppercase, or to lowercase and work in that case.

https://www.techonthenet.com/c_language/standard_library_functions/ctype_h/toupper.php

It'll save you from having to do double comparisons everywhere